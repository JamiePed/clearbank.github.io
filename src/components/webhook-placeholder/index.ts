export { default } from './webhook-placeholder'
