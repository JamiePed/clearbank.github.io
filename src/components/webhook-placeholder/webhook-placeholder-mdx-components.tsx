import React from 'react'

import * as DataRow from 'src/components/data-row/data-row.styles'

import CodeBlock from '../mdxComponents/codeBlock'
import AnchorTag from '../mdxComponents/anchor'
import * as Styles from '../mdxComponents/mdx-components.styles'

import defaultMDXComponents from '../mdxComponents/'

import '../styles.css'

export default {
  ...defaultMDXComponents,
  h1: (props) => (
    <Styles.H1
      as='h1'
      className='cannon'
      id={props.children.replace(/\s+/g, '').toLowerCase()}
      {...props}
    />
  ),
  h2: (props) => (
    <Styles.Header
      as='h2'
      className='trafalga'
      id={props.children.replace(/\s+/g, '').toLowerCase()}
      {...props}
    />
  ),
  h3: (props) => (
    <Styles.H3
      as='h3'
      className='paragon'
      id={props.children.replace(/\s+/g, '').toLowerCase()}
      {...props}
    />
  ),
  h4: (props) => (
    <DataRow.BlockTitle
      as='h4'
      className='double-pica'
      id={props.children.replace(/\s+/g, '').toLowerCase()}
      {...props}
    />
  ),
  h5: (props) => (
    <DataRow.Header
      as='h5'
      id={props.children.replace(/\s+/g, '').toLowerCase()}
      {...props}
    />
  ),
  h6: (props) => (
    <Styles.Header
      as='h6'
      className='long-primer'
      id={props.children.replace(/\s+/g, '').toLowerCase()}
      {...props}
    />
  ),
  p: props => <Styles.Paragraph as='p' className='body-copy' {...props} />,
  pre: props => <Styles.Pre {...props} />,
  code: CodeBlock,
  a: props => <Styles.Link {...props} />,
  ul: props => <Styles.List className='list' {...props} />,
  ol: props => <Styles.List as='ol' className='list' {...props} />,
  li: props => <Styles.ListItem {...props} />,
  img: props => <Styles.Img {...props} />
}
