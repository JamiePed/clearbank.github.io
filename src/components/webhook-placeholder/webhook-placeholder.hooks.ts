import { useStaticQuery, graphql } from 'gatsby'

export const getAllWebhooks = (): any => {
  const query = useStaticQuery(
    graphql`
      {
        allMdx(filter: { fields: {}, frontmatter: { webhook: { in: true } } }) {
          edges {
            node {
              fields {
                id
                title
              }
              fields {
                slug
                title
              }
              body
              frontmatter {
                version
                title
              }
            }
          }
        }
      }
    `
  )

  return query
}
