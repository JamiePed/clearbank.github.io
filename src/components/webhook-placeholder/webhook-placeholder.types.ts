export interface WebhookPlaceholderProps {
  render?: string
  title?: string
}
