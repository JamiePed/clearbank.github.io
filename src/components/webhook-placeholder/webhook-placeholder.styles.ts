import styled from 'styled-components'
import * as DataRow from 'src/components/data-row/data-row.styles'
import * as MDXComponents from '../mdxComponents/mdx-components.styles'
import * as Table from 'src/components/table/table.styles'

import { colors, breakpoints, fonts } from 'src/components/theme'
export const Container = styled.aside`

  ${MDXComponents.List} {
    margin-bottom: 5px;
  }

  ${MDXComponents.ListItem} {
    font-size: 16px;
    margin-bottom: 0;
  }

  ${DataRow.Description} {
    margin-bottom: 5px;
    font-size: 16px;
  }

  ${DataRow.Header} {
    margin-bottom: 0;
    margin-top: 30px;
  }

  ${DataRow.SubHeader} {
    margin-bottom: 20px;
  }

  // code samples
  ${MDXComponents.Pre} {
    margin: 15px 0 30px 0;
    max-width: 100%;
  }

  ${DataRow.BlockTitle} {
    margin-top: 30px;
    margin-bottom: 5px !important;
  }

  ${DataRow.BlockTitle} + ${DataRow.Header} {
    margin-top: 10px;
  }

  /* Table Stlyes */

  ${Table.Table} {
    margin-top: 15px;
    margin-bottom: 60px;
    width: 100%;
    table-layout: fixed; // take width from td/ths in first tr

    @media screen and (min-width: ${breakpoints.xLarge}) {
        table-layout: auto;
      }
  }

  ${Table.Thead} ${Table.Th} {
    width: auto;

    &:first-child {
      width: 40%;

      @media screen and (min-width: ${breakpoints.xLarge}) {
        width: auto;
      }
    }
  }

  // first-column
  ${Table.Tbody} ${Table.Trow} ${Table.Tdata}:first-child {
    font-size: 13px;
    font-weight: bold;
    font-family: ${fonts.monospace};
    color: ${colors.brandSecondaryDarkest};
  }

  /* Table: IF there are 3 columns, style the middle one. */

  ${Table.Tdata}:nth-last-of-type(2):nth-child(2) {
    font-size: 13px;
    color: ${colors.brandGreenDark};
    font-family: ${fonts.monospace};
  }

`

export const Link: any = styled.a``
export const Version: any = styled.p``
export const Title: any = styled.div``
