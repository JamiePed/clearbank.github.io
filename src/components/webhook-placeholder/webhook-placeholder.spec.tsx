import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './webhook-placeholder'

import * as Types from './webhook-placeholder.types'

jest.mock('__mocks__/gatsby-plugin-mdx')

let component: any = null
const defaultProps:Types.WebhookPlaceholderProps = {
  render: 'title',
  title: 'title'
}

beforeEach(() => {
  component = render(<Component {...defaultProps} />)
})


describe('Component', () => {
  test('matches snapshot', () => {
      expect(component.asFragment()).toMatchSnapshot()
  })
})
