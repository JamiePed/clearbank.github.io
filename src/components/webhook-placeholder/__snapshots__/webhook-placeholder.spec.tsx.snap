// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<DocumentFragment>
  .c0 .c5 {
  margin-bottom: 5px;
}

.c0 .c6 {
  font-size: 16px;
  margin-bottom: 0;
}

.c0 .c4 {
  margin-bottom: 5px;
  font-size: 16px;
}

.c0 .c2 {
  margin-bottom: 0;
  margin-top: 30px;
}

.c0 .c3 {
  margin-bottom: 20px;
}

.c0 .c7 {
  margin: 15px 0 30px 0;
  max-width: 100%;
}

.c0 .c1 {
  margin-top: 30px;
  margin-bottom: 5px !important;
}

.c0 .c1 + .c2 {
  margin-top: 10px;
}

.c0 .c8 {
  margin-top: 15px;
  margin-bottom: 60px;
  width: 100%;
  table-layout: fixed;
}

.c0 .c9 .c10 {
  width: auto;
}

.c0 .c9 .c10:first-child {
  width: 40%;
}

.c0 .c11 .c13 .c12:first-child {
  font-size: 13px;
  font-weight: bold;
  font-family: "Roboto Mono",monospace;
  color: #000000;
}

.c0 .c12:nth-last-of-type(2):nth-child(2) {
  font-size: 13px;
  color: #178880;
  font-family: "Roboto Mono",monospace;
}

@media screen and (min-width:1200px) {
  .c0 .c8 {
    table-layout: auto;
  }
}

@media screen and (min-width:1200px) {
  .c0 .c9 .c10:first-child {
    width: auto;
  }
}

<aside
    class="c0 page-menu-entry-parent"
  />
</DocumentFragment>
`;
